### PR TITLE
NamedFile: range upgrade

### PR DESCRIPTION
**Pull request by ghost**
Range header got fixed, from now on it should work as intended. Requested range is now inclusive. 

Range header in a response changed to Content-Range: Range is request only, Contene-Range is response only. [I should read more](https://tools.ietf.org/html/rfc7233)

Enabled Accept-Ranges since tests showed me the error messages didn't had any connection with this header. I found some other interesting stuff here, I'll open issues if I can't find the culprit.

Also added tests for content-range(new), content-length(placed in a different fn, added new tests), and the test for HTTP Status 416 was missing.

### CODE REVIEW

File: src/fs.rs [hunk 1]
```diff
@@ -300,26 +298,21 @@ impl Responder for NamedFile {
...
-                    resp.header(header::RANGE, format!("*/{}", length));
+                    resp.header(header::CONTENT_RANGE, format!("bytes */{}", length));
                     return Ok(resp.status(StatusCode::RANGE_NOT_SATISFIABLE).finish());
                 };
             } else {
                 return Ok(resp.status(StatusCode::BAD_REQUEST).finish());
             };
         };
 
+        resp.header(
```
Comment (fafhrd91) [inline src/fs.rs]: do we always need `content-range` header?

**Comment by ghost** [comment]
You're absolutely right. Modified.

File: src/fs.rs [hunk 2]
```diff
@@ -289,37 +289,35 @@ impl Responder for NamedFile {
                 resp.set(header::ETag(etag));
             });
 
-        // TODO: Debug, enabling "accept-ranges: bytes" causes problems with
-        // certain clients when not using the ranges header.
-        //resp.header(header::ACCEPT_RANGES, format!("bytes"));
+        resp.header(header::ACCEPT_RANGES, format!("bytes"));
```
Comment (fafhrd91) [inline src/fs.rs]: remove `format!`, `.header()` works with static str

**Comment by fafhrd91** [comment]
did you test ranges in browser?

**Comment by ghost** [comment]
Yes. Works as intended. Depends on the filetype too.

**Review by fafhrd91 — APPROVED** [review]

**Comment by ghost** [comment]
Thanks!

Fun fact: Does not work with all the browsers apparently: https://bugzilla.mozilla.org/show_bug.cgi?id=1418430

**Comment by fafhrd91** [comment]
thanks

**Pull request merged**